### PR TITLE
Contributor exports save

### PR DESCRIPTION
**Pull request by azime**

### CODE REVIEW

**Comment by datanel** [comment]
no test?

**Comment by datanel** [comment]
don't forget the swagger doc for get exports :wink:

**Comment by nberard** [comment]
i would move the GET export to __GET /contributors/<contributor_id>/exports__ and add a location headers in result of __POST /contributors/<contributor_id>/actions/export__
what do u think?

File: tartare/core/models.py [hunk 1]
```diff
@@ -468,3 +468,33 @@ class MongoJobSchema(Schema):
     step = fields.String(required=False)
     started_at = fields.DateTime(required=False)
     updated_at = fields.DateTime(required=False)
+
+
+class ContributorExports(object):
```
Comment (pbougue) [inline tartare/core/models.py]: Why plural? for the collection ok, but here it's a simple object, no?
(to be changed everywhere if agreed)

File: tartare/tasks.py [hunk 2]
```diff
@@ -96,9 +96,11 @@ def contributor_export(contributor, job):
         context = merge(contributor, context)
 
         models.Job.update(job_id=job.id, state="running", step="postprocess")
-        postprocess(contributor, context)
+        context = postprocess(contributor, context)
 
         models.Job.update(job_id=job.id, state="done")
+        # insert export in mango db
```
Comment (pbougue) [inline tartare/tasks.py]: typo: m`o`ngo

File: tests/integration/mongo/dataset_fetcher_test.py [hunk 3]
```diff
@@ -17,3 +21,16 @@ def test_fetch_data_from_input_failed(mocker):
         data_fetcher.fetch()
     except FileNotFoundError:
         pass
+
+
+class TestFtecher():
```
Comment (pbougue) [inline tests/integration/mongo/dataset_fetcher_test.py]: typo

File: tests/integration/mongo/dataset_fetcher_test.py [hunk 4]
```diff
@@ -17,3 +21,16 @@ def test_fetch_data_from_input_failed(mocker):
...
+    @mock.patch('urllib.request.urlretrieve', side_effect=mock_urlretrieve)
+    def test_fetcher(self, urlretrieve_func):
+        data_source = DataSource(666, 'myDS', 'gtfs', {"type": "ftp", "url": "bob"})
```
Comment (pbougue) [inline tests/integration/mongo/dataset_fetcher_test.py]: a readable id would be better than `666` (`myDSid` ?)

File: tartare/core/models.py [hunk 5]
```diff
@@ -468,3 +468,33 @@ class MongoJobSchema(Schema):
...
+
+class ContributorExport(object):
+    mongo_collection = 'contributor_exports'
+
+    def __init__(self, contributor_id, gridfs_id, data_sources=None):
+        self.id = str(uuid.uuid4())
+        self.contributor_id = contributor_id
+        self.gridfs_id = gridfs_id
+        self.created_at = datetime.utcnow()
+        self.data_sources = [] if data_sources is None else data_sources
```
Comment (nberard) [inline tartare/core/models.py]: why is data_sources an array?

File: tartare/core/models.py [hunk 6]
```diff
@@ -468,3 +468,33 @@ class MongoJobSchema(Schema):
...
+        raw = mongo.db[cls.mongo_collection].find({'contributor_id': contributor_id}).sort("created_at", -1)
+        return MongoContributorExportSchema(many=True).load(raw).data
+
+
+class MongoContributorExportSchema(Schema):
+    id = fields.String(required=True, load_from='_id', dump_to='_id')
+    contributor_id = fields.String(required=True)
+    gridfs_id = fields.String(required=True)
+    created_at = fields.DateTime(required=True)
+    data_sources = fields.List(fields.String())
```
Comment (nberard) [inline tartare/core/models.py]: why is data_sources a list?

Comment (azime) [inline tartare/core/models.py]: In the case of merge

Comment (nberard) [inline tartare/core/models.py]: ok

**Pull request merged**